src/Sportradar.OddsFeed.SDK/Entities/REST/Internal/MarketNames/EntityNameExpression.cs

```diff
@@ -107,13 +107,21 @@ private async Task<string> GetHomeCompetitor(CultureInfo culture)
             if (_sportEvent is IMatch match)
             {
                 var competitor = await InvokeAndWrapAsync(match.GetHomeCompetitorAsync).ConfigureAwait(false);
+                await competitor.EnsureProfileLoaded();
                 return competitor?.GetName(culture);
             }
 
             if (_sportEvent is IStage stage)
             {
                 var competitors = await InvokeAndWrapAsync(stage.GetCompetitorsAsync).ConfigureAwait(false);
-                return competitors?.First().GetName(culture);
+                var competitor = competitors?.First();
```

Comment:
Changes made here were done to make sure it is using the `EnsureProfileLoaded` method and maintain the previous functionality.